Framework/DataHandling/src/LoadMask.cpp

```diff
@@ -39,6 +39,232 @@ using namespace Mantid::Kernel;
 using namespace Mantid::API;
 using namespace std;
 
+namespace {
+// service routines
+//-------------------------------------------------------------------------------------------
+/** Convert ranged vectors to single-valued vector
+* @param singles -- input vector of single numbers to copy to result without
+*                   changes.
+* @param ranges  -- input vector of data ranges -- pairs of min-max values,
+*                   expanded to result as numbers from min to max
+*                   inclusively, with step 1
+* @param tot_singles -- on input contains range of single values already
+*                   copied into the result by previous call to the routine,
+*                   on output, all singles and expanded pairs
+*                   from the input are added to it.
+*/
+template <typename T>
+void convertToVector(const std::vector<T> &singles,
+                     const std::vector<T> &ranges,
+                     std::vector<T> &tot_singles) {
+
+  // find the size of the final vector of masked values
+  size_t n_total(singles.size() + tot_singles.size());
+  for (size_t i = 0; i < ranges.size(); i += 2) {
+    n_total += ranges[i + 1] - ranges[i] + 1;
+  }
+  // reserve space for all masked spectra
+  // for efficient memory operations
+  tot_singles.reserve(n_total);
+  // add singles to the existing singles
+  tot_singles.insert(tot_singles.end(), singles.begin(), singles.end());
+  // expand pairs
+  for (size_t i = 0; i < ranges.size(); i += 2) {
+    for (T obj_id = ranges[i]; obj_id < ranges[i + 1] + 1; ++obj_id) {
+      tot_singles.push_back(obj_id);
+    }
+  }
+}
+
+/*
+* Parse index range text to singles and pairs
```

Comment:
Are these supposed to be doxygen comments? Should they not start with `/**` instead of `/*` to be recognized as such?